app/src/main/java/com/example/ncov19traking/NCoVRecyclerAdapter.kt

```diff
@@ -16,38 +16,103 @@ import kotlin.collections.ArrayList
 class NCoVRecyclerAdapter : RecyclerView.Adapter<NCoVRecyclerAdapter.ViewHolder>(), Filterable {
 
     private val dataset = ArrayList<NumbersByCountry>()
+
     private var datasetForSearch = dataset
 
+    private var countryFilter: Filter = object : Filter() {
+        override fun performFiltering(constraint: CharSequence?): FilterResults {
+            return FilterResults().apply {
+                values = datasetForSearch.takeIf { constraint.isNullOrEmpty() } ?: let {
+                    val filterPattern = constraint.toString().toLowerCase(Locale.ROOT).trim()
+                    datasetForSearch.filter { it.country.contains(filterPattern, true) }
+                }
+            }
+        } /*{
+            var filteredList = ArrayList<NumbersByCountry>()
+            if (constraint.isNullOrEmpty()) {
+                filteredList = datasetForSearch
+            } else {
+                val filterPattern = constraint.toString().toLowerCase(Locale.ROOT).trim()
+
+                for (item in datasetForSearch) {
+                    if (item.country.toLowerCase(Locale.ROOT).contains(filterPattern)) {
+                        filteredList.add(item)
+                    }
+                }
+            }
+            val result = FilterResults()
+            result.values = filteredList
+            return result
+        }*/
```

Comment:
let's remove this code, since you commited it once if you want to get back to it you could always use git history to have it back.

Reply:
removed